### PR TITLE
Implement some Python best practices

### PR DESCRIPTION
**Pull request by aaugustin**
Improve source code quality with black, isort & flake8.

### CODE REVIEW

**Comment by aaugustin** [comment]
This is in reasonably good shape. There are two tasks left:

1. Figure out why isort doesn't behave the same locally and on Circle CI. Most likely this has to do with how it classifies first-party / third-party / unknown libraries. Then enable in on Circle CI.
2. Fix flake8 F841 warnings and stop ignoring them.

Assuming tests pass, I think it would be best to merge this PR and deal with these two items later. I'd like to do the repository structure changes first, so we're done with the large changes.

File: .circleci/config.yml [hunk 1]
```diff
@@ -90,12 +90,15 @@ jobs:
...
             - image: circleci/python:3.6
-        resource_class: small
+        resource_class: medium
         parallelism: 1
         steps:
             - checkout
-            - run: sudo pip install black
+            - run: sudo pip install --editable .
+            - run: sudo pip install torch tensorflow tensorboardX scikit-learn
+            - run: sudo pip install black git+git://github.com/timothycrosley/isort.git@e63ae06ec7d70b06df9e528357650281a3d3ec22#egg=isort
```
Comment (julien-c) [inline .circleci/config.yml]: shoud we ask them politely for a new release?

Comment (aaugustin) [reply .circleci/config.yml]: A couple others asked already. I'm not sure this would make a big difference.

**Comment by aaugustin** [comment]
`pyproject.toml` is supposed to be The Future (tm). `setup.cfg`was supposed to be The Future as well. That didn't quite work, but it gained some support.

Unfortunately, in the Python packaging ecosystem, some futures never became the present. That's why I'm conservative, at least until this changes: https://packaging.python.org/specifications/distribution-formats/

I'm happy to try converting setup.py to pyproject.toml if you're feeling adventurous. Let me know.

We can move the isort configuration there, but not the flake8 configuration until [this PR](https://gitlab.com/pycqa/flake8/issues/428) is merged.

**Comment by julien-c** [comment]
Alright thanks for the context!

**Pull request merged**